airbyte-integrations/connectors/source-snowflake/source_snowflake/schema_builder.py

```diff
@@ -0,0 +1,44 @@
+from typing import Dict
+
+
+class SchemaTypes:
+    string: Dict = {"type": ["null", "string"]}
+    number: Dict = {"type": ["null", "number"]}
+    integer: Dict = {"type": ["null", "number"]}
+    boolean: Dict = {"type": ["null", "boolean"]}
+    date: Dict = {"type": ["null", "string"], "format": "date"}
+    array_with_any: Dict = {"type": ["null", "array"], "items": {}}
+    object: Dict = {"type": ["null", "object"]}
+
+
+mapping_snowflake_type_airbyte_type = {
+    # Numeric data types
+    'float': SchemaTypes.number,
+    'number': SchemaTypes.integer,
+    'boolean': SchemaTypes.boolean,
+
+    # string data
+    'varchar': SchemaTypes.string,
+    'char': SchemaTypes.string,
+    'character': SchemaTypes.string,
+    'string': SchemaTypes.string,
+    'text': SchemaTypes.string,
+    'binary': SchemaTypes.string,
+    'varbinary': SchemaTypes.string,
+
+    # Date and time
+    'timestamp_tz': SchemaTypes.string,
+    'timestamp_ltz': SchemaTypes.string,
+    'timestamp_ntz': SchemaTypes.string,
+    'time': SchemaTypes.string,
+    'date': SchemaTypes.date,
+
+    # semi structured objects
+    'array': SchemaTypes.array_with_any,
+    'object': SchemaTypes.object,
+
+    # Not present in documentation
```

Comment:
That is kind of worrisome. Where did you see this? Should we contact Snowflake on this and ask them an exhaustive list of the types?

Reply:
The types are present in this link: https://docs.airbyte.com/integrations/destinations/snowflake#data-type-map

Here is the exhaustive list of data types in snowflake
https://docs.snowflake.com/en/sql-reference/intro-summary-data-types

Reply:
Can you printscreen that for me because "Find in page" using `fixed` in [this page](https://docs.snowflake.com/en/sql-reference/intro-summary-data-types) didn't yield results

Reply:
I received the type fixed when I was doing some tests on integers. You can see that in this screen shot.
I found this documentation about the topic: https://docs.snowflake.com/en/sql-reference/data-types-numeric
Is this making sense for you ? 
![image](https://github.com/artefactory/airbyte/assets/163012956/6de0e451-9ebe-4a33-97e5-ea9dd6d1ab4a)

Reply:
This is interesting and still worrisome haha! Can we document the reasoning as to how to create a table with these types and why we think it is an integer? Based on your printscreen, I'm still not sure if it should be `integer` or `number`

Reply:
Thank you for your comment. I think it should be considered as a number and not an integer since it it gives you the choice to set the precision on the decimal for the number. What is worrisome is that I found this new type randomly. It has not been listed as types of data in the snowflake summary. https://docs.snowflake.com/en/sql-reference/intro-summary-data-types